app/controllers/api/functions.php

```diff
@@ -1290,3 +1296,231 @@
+        }
+
+        $filterQueries = Query::groupByType($queries)['filters'];
+
+        $response->dynamic(new Document([
+            'variables' => $dbForProject->find('variables', $queries),
+            'total' => $dbForProject->count('variables', $filterQueries, APP_LIMIT_COUNT),
+        ]), Response::MODEL_VARIABLE_LIST);
+    });
+
+App::get('/v1/functions/:functionId/variables/:variableId')
+    ->desc('Get Variable')
+    ->groups(['api', 'functions'])
+    ->label('scope', 'functions.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'getVariable')
+    ->label('sdk.description', '/docs/references/functions/get-variable.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_VARIABLE)
+    ->param('functionId', null, new UID(), 'Function unique ID.', false)
+    ->param('variableId', null, new UID(), 'Variable unique ID.', false)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $functionId, string $variableId, Response $response, Database $dbForProject) {
+        $function = $dbForProject->getDocument('functions', $functionId);
+
+        if ($function->isEmpty()) {
+            throw new Exception(Exception::FUNCTION_NOT_FOUND);
+        }
+
+        $variable = $dbForProject->findOne('variables', [
+            Query::equal('_uid', [$variableId]),
+            Query::equal('functionInternalId', [$function->getInternalId()]),
+        ]);
+
+        if ($variable === false || $variable->isEmpty()) {
+            throw new Exception(Exception::VARIABLE_NOT_FOUND);
+        }
+
+        $response->dynamic($variable, Response::MODEL_VARIABLE);
+    });
+
+App::put('/v1/functions/:functionId/variables/:variableId')
+    ->desc('Update Variable')
+    ->groups(['api', 'functions'])
+    ->label('scope', 'functions.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'updateVariable')
+    ->label('sdk.description', '/docs/references/functions/update-variable.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_VARIABLE)
+    ->param('functionId', null, new UID(), 'Function unique ID.', false)
+    ->param('variableId', null, new UID(), 'Variable unique ID.', false)
+    ->param('key', null, new Text(255), 'Variable key. Max length: 255 chars.', true)
+    ->param('value', null, new Text(8192), 'Variable value. Max length: 8192 chars.', true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $functionId, string $variableId, ?string $key, ?string $value, Response $response, Database $dbForProject) {
+        if (empty($key) && empty($value)) {
```

Comment:
```suggestion
        if (empty($key) || empty($value)) {
```

should be like this right?

Reply:
Doing `OR` would mean you need to provide both. What we check here is that at least one is provided. Notice this is an update endpoint, not create. I think the original is right. Right?

Reply:
I think this should be something like this 
```suggestion
        if (empty($key) || $value === null) {
```

Reply:
yeah, should be required 👍🏻

Reply:
`key` is now required. Possible combinations:
- no input: 400
- value only: 400
- key only: 200, key updated, value the same
- key + value: 200, both updated

All cases got their own tests.